paper/src/main/java/com/denizenscript/denizen/paper/events/PlayerSetSpawnScriptEvent.java

```diff
@@ -0,0 +1,108 @@
+package com.denizenscript.denizen.paper.events;
+
+import com.denizenscript.denizen.events.BukkitScriptEvent;
+import com.denizenscript.denizen.objects.LocationTag;
+import com.denizenscript.denizen.utilities.implementation.BukkitScriptEntryData;
+import com.denizenscript.denizencore.objects.ObjectTag;
+import com.denizenscript.denizencore.objects.core.ElementTag;
+import com.denizenscript.denizencore.scripts.ScriptEntryData;
+import com.destroystokyo.paper.event.player.PlayerSetSpawnEvent;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import static com.denizenscript.denizen.paper.PaperModule.componentToJson;
+import static com.denizenscript.denizen.paper.PaperModule.jsonToComponent;
+
+public class PlayerSetSpawnScriptEvent extends BukkitScriptEvent implements Listener {
+
+    // <--[event]
+    // @Events
+    // player spawn changes
+    //
+    // @Cancellable true
+    //
+    // @Location true
+    //
+    // @Plugin Paper
+    //
+    // @Group Paper
+    //
+    // @Triggers when a player's spawn point changes.
+    //
+    // @Switch cause:<cause> to only process when the cause for the event matches the input.
+    //
+    // @Context
+    // <context.cause> returns the reason the player's spawn point changed. A list of causes can be found at <@link url https://jd.papermc.io/paper/1.21.5/com/destroystokyo/paper/event/player/PlayerSetSpawnEvent.Cause.html>.
+    // <context.forced> returns whether this event will persist through source block (bed or respawn anchor) removal.
+    // <context.location> returns a LocationTag of the new respawn location, if any.
+    // <context.message> returns the notification message that is sent to the player.
+    // <context.notify> returns whether the player will be notified their spawn point changed.
+    //
+    // @Determine
+    // "FORCED:<ElementTag(Boolean)>" to set whether the player's new spawnpoint will persist even if the bed or respawn anchor that triggered the event is removed.
+    // "MESSAGE:<ElementTag>" to set the notification message that is sent to the player.
+    // "NOTIFY:<ElementTag(Boolean)>" to set whether the player will be notified their spawnpoint changed.
+    // LocationTag to change the respawn location.
+    //
+    // @Player Always.
+    //
+    // -->
+
+    public PlayerSetSpawnScriptEvent() {
+        registerCouldMatcher("player spawn changes");
+        registerSwitches("cause");
+        this.<PlayerSetSpawnScriptEvent, ElementTag>registerOptionalDetermination("forced", ElementTag.class, (evt, context, value) -> {
+            if (value.isBoolean()) {
+                evt.event.setForced(value.asBoolean());
+                return true;
+            }
+            return false;
+        });
+        this.<PlayerSetSpawnScriptEvent, ElementTag>registerDetermination("message", ElementTag.class, (evt, context, message) -> {
+            evt.event.setNotification(jsonToComponent(message.toString()));
```

Comment:
A. generally directly referencing the method is preferred over static imports
B. why are you using the JSON methods? this is meant to convert between the Denizen advanced text syntax and Components, you'd want `PaperModule.parseFormattedText(meesage.asString())`